File: Recursion/CheckSortedOrNot.cpp
#include <iostream>
using namespace std;

bool isSorted(int *arr, int size)
{
    if (size <= 1)
        return true;
    if (arr[0] > arr[1])
        return false;
    else
    {
        bool ans = isSorted(arr + 1, size - 1);
        return ans;
    }
}

int main()
{
    int arr[5] = {1, 2, 6, 4, 5};
    int size = 5;
    bool ans = isSorted(arr, size);
    if (ans)
        cout << "Array is sorted" << endl;
    else
        cout << "Array is not sorted" << endl;
    return 0;
}

File: Recursion/CheckSortedOrNot.java
package arnab.DSA.Recursion;

public class CheckSortedOrNot {
    public static void main(String[] args) {
        int[] arr = { 1, 2, 4, 8, 7 };
        // int size = 5;
        boolean ans = isSortedArray(arr, 1, arr.length);
        if (ans) {
            System.out.println("Sorted");
        } else {
            System.out.println("Not sorted");
        }
    }

    static boolean isSortedArray(int[] arr, int indx, int size) {
        // base condition
        if (size <= 1) {
            return true;
        }
        if (arr[indx - 1] > arr[indx]) {
            return false;
        } else {
            Boolean ans = isSortedArray(arr, indx + 1, size - 1);
            return ans;
        }
    }
}
